### DISCUSSION TITLE
[Feature Request] 图标库分离或者增加插槽

### DISCUSSION THREAD

**Discussion by carlosyan1807** [element-plus/element-plus#1178] [Ideas]
- 内置图标数量有限，不可能满足所有需要，使用自定义图标成常态
- 扩大内置图标数量必然增大包
- 一些组件中的图标位没有插槽，是写死的，比较难替换
- 自定义图标库的写法太多了，写成component的，svg的，js的或者直接引入一些第三方，都要去替换ep的css么

举两个例子，Collapse和Tabs
![image](https://user-images.githubusercontent.com/50797685/103435230-e9458080-4c46-11eb-93cc-32f3803a7def.png)
![image](https://user-images.githubusercontent.com/50797685/103435240-0712e580-4c47-11eb-850d-0f25c44a1d5a.png)

期望
- 内置图标库是不是可以分离出去，有需要的可以单独引入，当然想开箱即用肯定还得有内置。。。😂
- 组件所有图标位增加插槽，不管是不是自定义图标，就算使用内置的也能方便替换啊

正在尝试从ant-design-vue转到ep，发现有些小地方还真是挺麻烦的

**Comment by zazzaz** [comment]
we are going to replace the built-in font icon with separate svg icons, you can check the progress here https://github.com/element-plus/element-plus-icons

**Reply by carlosyan1807** [reply]
that's great.

**Comment by zazzaz** [comment]
PR: https://github.com/element-plus/element-plus/pull/1153
